### PR TITLE
etcdmain: New logger

### PR DESCRIPTION
**Pull request by barakmich**
The first step in #2398 is to use the new logger I wrote.

Fortunately, it wraps log. So it's trivial to get started -- this is the minimal change (with new flags available for detailed logging in the future) in order to start fixing the log situation.

### CODE REVIEW

File: etcdmain/etcd.go [hunk 1]
```diff
@@ -389,3 +393,19 @@ func identifyDataDirOrDie(dir string) dirType {
...
+func setupLogging(cfg *config) {
+	capnslog.SetGlobalLogLevel(capnslog.INFO)
+	if cfg.debug {
+		capnslog.SetGlobalLogLevel(capnslog.DEBUG)
+	}
+	if cfg.logPkgs != "" {
+		repoLog := capnslog.MustRepoLogger("github.com/coreos/etcd")
+		settings, err := repoLog.ParseLogLevelConfig(cfg.logPkgs)
+		if err != nil {
+			log.Warningf("Couldn't parse log level string: %s, continuing with default levels", err.Error())
```
Comment (xiang90) [inline etcdmain/etcd.go]: do we need to attach pkg name as line 63?

Comment (barakmich) [inline etcdmain/etcd.go]: Nah. `repoLogger` is the handle to the set of all loggers in the repository, where `log` is the single one (`etcdmain` in this case) and all Formatters know which package a line came from.

The upgrade path won't touch repoLogger -- merely creating a `var log = capnslog.NewPackageLogger("github.com/coreos/etcd", "$PACKAGENAME")` in place of importing `log`. Even then, everything should still work -- and individual log calls can change their level (to log.Fatal, or log.Warning, for example)

Comment (xiang90) [inline etcdmain/etcd.go]: so we should probably remove the prefix at line 63?

Comment (barakmich) [inline etcdmain/etcd.go]: Ah, yes. That would make sense. Fixing.

**Comment by xiang90** [comment]
@barakmich I would like to see an example of  how `log-packages` flag should be used. Other than that, LGTM

**Comment by xiang90** [comment]
@barakmich Oh. We need to change the commit message though. Prefix with pkg name.

**Comment by luan** [comment]
Any chance of this getting ported over to 2.0? The debug logging would be really handy for us.

**Pull request merged**